### PR TITLE
Fix #21027

### LINKED ISSUE #21027: kube-up on GCE creates 3x the number of nodes expected
Spinning up a new GCE cluster, I get 9 nodes in 3 instance groups, but the validate-cluster call is only checking for 4 nodes (presumably one instance group's 3 nodes plus the master):

`Detected 5 ready nodes, found 5 nodes out of expected 4. Found more nodes than expected, your cluster may not behave correctly.`

Once 4+ nodes are found it happily continues and says that the cluster is running. But if you immediately run `kubectl get nodes` there are multiple nodes in the `NotReady` state, which should block validate-cluster from completing successfully.

### PR DESCRIPTION
**Pull request by gmarek**
Fixes #21027

### CODE REVIEW

**Comment by roberthbailey** [comment]
Thanks. LGTM.

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -708,7 +708,7 @@ function create-autoscaler() {
     local min_instances_per_mig=$(((${AUTOSCALER_MIN_NODES} + ${NUM_MIGS} - 1) / ${NUM_MIGS}))
     local last_min_instances=$((${AUTOSCALER_MIN_NODES} - (${NUM_MIGS} - 1) * ${min_instances_per_mig}))
 
-    for i in $(seq $((${NUM_MIGS} - 1))); do
+    for ((i=1; i<=$((${NUM_MIGS} - 1)); i++)); do
```
Comment (quinton-hoole) [inline cluster/gce/util.sh]: nit: Isn't "i<$NUM_MIGS" equivalent and simpler?  Same in above loop on line 650.

Comment (gmarek) [inline cluster/gce/util.sh]: Right.

**Comment by roberthbailey** [comment]
Still LGTM.

**Comment by roberthbailey** [comment]
@gmarek A bunch of e2e tests failed. Can you see if they were flakes or issues with this PR?

**Comment by gmarek** [comment]
It'a s flake, one node failed:
```
Feb 12 08:15:25.833: FAIL: All nodes should be ready after test, Not ready nodes: [{{ } {e2e-gce-builder-3-2-minion-u3ke   /api/v1/nodes/e2e-gce-builder-3-2-minion-u3ke 063a86be-d1a3-11e5-be9a-42010af00002 6269 0 2016-02-12 08:09:41 -0800 PST <nil> <nil> map[failure-domain.alpha.kubernetes.io/region:us-central1 failure-domain.alpha.kubernetes.io/zone:us-central1-f kubernetes.io/hostname:e2e-gce-builder-3-2-minion-u3ke] map[]} {10.245.4.0/24 6418528712169776226 gce://kubernetes-jenkins-pull/us-central1-f/e2e-gce-builder-3-2-minion-u3ke false} {map[cpu:{2.000 DecimalSI} memory:{7864139776.000 BinarySI} pods:{110.000 DecimalSI}] map[cpu:{2.000 DecimalSI} memory:{7864139776.000 BinarySI} pods:{110.000 DecimalSI}]  [{OutOfDisk Unknown 2016-02-12 08:13:38 -0800 PST 2016-02-12 08:14:22 -0800 PST NodeStatusUnknown Kubelet stopped posting node status.} {Ready Unknown 2016-02-12 08:13:38 -0800 PST 2016-02-12 08:14:22 -0800 PST NodeStatusUnknown Kubelet stopped posting node status.}] [{InternalIP 10.240.0.6} {ExternalIP 130.211.159.57}] {{10250}} { 1156505C-B45E-5955-D848-231171A69AA1 cce001e1-f0de-4919-8b37-2ab188f36c29 3.16.0-4-amd64 Debian GNU/Linux 7 (wheezy) docker://1.9.1 v1.2.0-alpha.7.703+1ad9da80cce6d2 v1.2.0-alpha.7.703+1ad9da80cce6d2} [{[gcr.io/google_containers/kube-proxy:fb88af1d1edd8e5ef46464941b5051bb] 164509573} {[nginx:latest] 134581595} {[gcr.io/google_containers/fluentd-elasticsearch:1.14] 562034622} {[redis:latest] 151289204} {[gcr.io/google_containers/busybox:1.24 busybox:latest] 1113554} {[gcr.io/google_containers/hostexec:1.2] 13209617} {[gcr.io/google_containers/mounttest:0.5] 1718853} {[gcr.io/google_containers/pause:2.0] 350164} {[gcr.io/google_containers/nettest:1.6] 10374838} {[gcr.io/google_containers/pause:0.8.0] 241656} {[gcr.io/google_containers/eptest:0.1] 2970692} {[gcr.io/google_containers/serve_hostname:1.1] 4522409} {[gcr.io/google_containers/nginx:1.7.9] 91664166} {[gcr.io/google_containers/test-webserver:latest] 4534272}]}}]
```
cc @kubernetes/goog-node 

@k8s-bot test this

**Comment by gmarek** [comment]
@k8s-bot test this issue: #21245

**Comment by gmarek** [comment]
One node was down...

@k8s-bot test this issue: #IGNORE

**Comment by gmarek** [comment]
@k8s-bot unit test this issue: #21257

**Pull request merged**